lib/suggestion-list.coffee

```diff
@@ -0,0 +1,112 @@
+{Model} = require 'theorist'
+{Emitter, CompositeDisposable} = require 'event-kit'
+
+module.exports =
+class SuggestionList extends Model
+  constructor: ->
+    @emitter = new Emitter
+    @subscriptions = new CompositeDisposable
+    # Allow keyboard navigation of the suggestion list
+    @subscriptions.add(atom.commands.add 'autocomplete-suggestion-list',
+      'autocomplete-plus:confirm': @confirmSelection,
+      'autocomplete-plus:select-next': @selectNext,
+      'autocomplete-plus:select-previous': @selectPrevious,
+      'autocomplete-plus:cancel': @cancel
+    )
+
+  addKeyboardInteraction: ->
+    @removeKeyboardInteraction()
+    keys =
+      'escape': 'autocomplete-plus:cancel'
+
+    completionKey = atom.config.get('autocomplete-plus.confirmCompletion') || ''
+    navigationKey = atom.config.get('autocomplete-plus.navigateCompletions') || ''
+
+
+    keys['tab'] = 'autocomplete-plus:confirm' if completionKey.indexOf('tab') > -1
+    keys['enter'] = 'autocomplete-plus:confirm' if completionKey.indexOf('enter') > -1
+
+    if @items?.length > 1 and navigationKey == 'up,down'
+      keys['up'] =  'autocomplete-plus:select-previous'
+      keys['down'] = 'autocomplete-plus:select-next'
+    else
+      keys['ctrl-n'] = 'autocomplete-plus:select-next'
+      keys['ctrl-p'] = 'autocomplete-plus:select-previous'
+
+    @keymaps = atom.keymaps.add(
```

Comment:
This is new, created by @park9140 specifically to address #173. It also allows us to simplify the way end-users achieve what is probably the # 1 issue category in this repo: "how do I change the keybindings". That has dropped off precipitously since this was added. We ended up backing out some changes to ensure we can still support user keymaps.

That said, yes - it would be great to evolve this further!